src/mount/projfs.rs

```diff
@@ -0,0 +1,529 @@
+
+            buf[bytes_written..(bytes_written + bytes_pending)]
+                .copy_from_slice(&current_chunk[0..bytes_pending]);
+            bytes_written += bytes_pending;
+
+            if bytes_pending == current_chunk.len() {
+                let _ = self.iter.next();
+            } else {
+                *current_chunk = current_chunk.slice(bytes_pending..);
+            }
+        }
+
+        Ok(bytes_written)
+    }
+}
+
+const UNIX_WIN_DIFF_SECS: i64 = 11644473600;
+fn unix_time_to_windows(unix_seconds: i64, unix_nanos: u32) -> u64 {
+    if unix_seconds < -UNIX_WIN_DIFF_SECS {
+        return 0;
+    }
+
+    let win_seconds = (unix_seconds + UNIX_WIN_DIFF_SECS) as u64;
+    win_seconds * 1_000_000_000 / 100 + (unix_nanos / 100) as u64
+}
+
+/* https://learn.microsoft.com/en-us/windows/win32/fileio/file-attribute-constants */
+const FILE_ATTRIBUTE_READONLY: u32 = 0x00000001;
+const FILE_ATTRIBUTE_DIRECTORY: u32 = 0x00000010;
+const FILE_ATTRIBUTE_NOT_CONTENT_INDEXED: u32 = 0x00002000;
+const FILE_ATTRIBUTE_RECALL_ON_OPEN: u32 = 0x00040000;
+
+/* Note: Using FILE_ATTRIBUTE_READONLY on directories will cause the explorer to *always* list all second level subdirectory entries */
+const DIRECTORY_ATTRIBUTES: u32 =
+    FILE_ATTRIBUTE_DIRECTORY | FILE_ATTRIBUTE_NOT_CONTENT_INDEXED | FILE_ATTRIBUTE_RECALL_ON_OPEN;
+
+fn index_entry_to_directory_entry(entry: &IndexEntry) -> Option<DirectoryEntry> {
+    let file_name = entry.apath.split('/').last()?;
+    if entry.kind == Kind::Dir {
+        Some(
+            DirectoryInfo {
+                directory_name: file_name.to_string(),
+                directory_attributes: DIRECTORY_ATTRIBUTES,
+
+                /* currently conserve does not differentiate between the different time stamps */
+                creation_time: unix_time_to_windows(entry.mtime, entry.mtime_nanos),
+                last_access_time: unix_time_to_windows(entry.mtime, entry.mtime_nanos),
+                last_write_time: unix_time_to_windows(entry.mtime, entry.mtime_nanos),
+            }
+            .into(),
+        )
+    } else if entry.kind == Kind::File {
+        Some(
+            FileInfo {
+                file_name: file_name.to_string(),
+                file_size: entry.addrs.iter().map(|block| block.len).sum(),
+                file_attributes: FILE_ATTRIBUTE_READONLY,
+
+                /* currently conserve does not differentiate between the different time stamps */
+                creation_time: unix_time_to_windows(entry.mtime, entry.mtime_nanos),
+                last_access_time: unix_time_to_windows(entry.mtime, entry.mtime_nanos),
+                last_write_time: unix_time_to_windows(entry.mtime, entry.mtime_nanos),
+            }
+            .into(),
+        )
+    } else {
+        None
+    }
+}
+
+struct ArchiveProjectionSource {
+    archive: Archive,
+
+    stored_tree_cache: Mutex<LruCache<BandId, Arc<StoredTree>>>,
```

Comment:
Probably later we should move these out of this module into some kind of generic read cache for the archive; maybe into the `Archive` itself.

Reply:
That sounds usefull.  
Maybe an abstraction of an Archive, e.g. `CachedArchive`?